snapdtool/cmdutil.go

```diff
@@ -112,9 +112,33 @@ func CommandFromSystemSnap(name string, cmdArgs ...string) (*exec.Cmd, error) {
 	cmdPath := filepath.Join(root, name)
 
 	if from == "snapd" {
-		// We do not need to specify the linker or library
-		// path from files coming from the snapd snap.
-		return exec.Command(cmdPath, cmdArgs...), nil
+		// check if the elf interpreter invoked by the binary will work
+		// without any tweaks, which is true if the snaps are mounted at
+		// /snap, otherwise we need to set up a command to invoke it
+		// directly
+		// TODO perhaps check the symlink right away
+		slashSnapPrefix := filepath.Join(dirs.GlobalRootDir, "snap") + "/"
+		if strings.HasPrefix(root, slashSnapPrefix) {
```

Comment:
Can we check also in the condition that  `filepath.Join(dirs.GlobalRootDir, "snap/snapd/current")` does not exist (following symlinks). So that if a build uses /var/lib something in `SnapMountDir`, but the installation has a functional /snap symlink, then we do not use `ld --library-path`.

An in my opinion, we should have a warning, if the symlink does not exist.

Reply:
This has been resolved through https://github.com/canonical/snapd/pull/14375